core/src/main/scala/com/yahoo/maha/core/query/presto/PrestoQueryGenerator.scala

```diff
@@ -491,15 +491,24 @@ class PrestoQueryGenerator(partitionColumnRenderer:PartitionColumnRenderer, udfS
       } else {
         "LEFT OUTER JOIN"
       }
-      // pkColName ?? alias ?? cc3_id
+
+      // Columns on which join is done must be of the same type. Add explicit CAST otherwise.
+      val joinCondition = {
+        if (fkCol.dataType.getClass.equals(dimBundle.dim.columnsByNameMap(pkColName).dataType.getClass)) {
```

Comment:
dimBundle.dim.columnsByNameMap(pkColName).dataType is unsafe, put require and make sure map has value

Reply:
Added require